### DISCUSSION TITLE
Receiving duplicates in realtime database changes

### DISCUSSION THREAD

**Discussion by sam26880** [supabase/supabase#13908] [Questions]
I have a simple page where I'm inserting a message into a message table and then on the frontend detecting it. I have used the sample code provided.

However, everytime, I seem to be detecting 3 events for every insert.

Any ideas?

I have posted this on stackoverflow as well with some more code details
https://stackoverflow.com/questions/76068621/supabase-realtime-detecting-duplicates-for-the-same-event

**Comment by GaryAustin1** [comment]
Does your console.log for subscribe fire 3 times?
You need to console.log to make sure you onMount is not firing 3 times.
Also are you getting the insert from your subscribe event, or from hand inserting into the table?  Usually right after subscribe status the new version is still not quite ready for postgres events and you have to wait for a status even.  I don't see how this would cause 3 events though.
https://supabase.com/docs/guides/realtime/protocol#system-messages
